### PR TITLE
refactor: `curve-commons` package

### PR DESCRIPTION
**Pull request by DanielSchiavini**
image

- create curve-commons package to keep code shared between the apps
- move first feature onboard-helpers into a connect-wallet feature
- In the future we will add more features and widgets to this package

### CODE REVIEW

File: apps/main/next.config.js [hunk 1]
```diff
@@ -27,7 +27,7 @@ const nextConfiguration = {
     unoptimized: true,
   },
   trailingSlash: true,
-  transpilePackages: ['onboard-helpers', 'ui'],
+  transpilePackages: ['curve-features', 'curve-ui-kit'],
```
Comment (DanielSchiavini) [inline apps/main/next.config.js:30]: is this even needed at all?

File: .eslintrc.js [hunk 2]
```diff
@@ -4,7 +4,7 @@ module.exports = {
   extends: ['custom'],
   settings: {
     next: {
-      rootDir: ['apps/*/', 'packages/ui/*/', 'packages/onboard-helpers/*/', 'packages/curve-ui-kit/*/'],
+      rootDir: ['apps/*/', 'packages/ui/*/', 'packages/features/*/', 'packages/curve-ui-kit/*/'],
```
Comment (ignromanov) [inline .eslintrc.js:7]:
```suggestion
      rootDir: ['apps/*/', 'packages/ui/*/', 'packages/curve-features/*/', 'packages/curve-ui-kit/*/'],
```

File: apps/main/next.config.js (hunk 1 above)
Comment (ignromanov) [reply apps/main/next.config.js:30]: I guess yes, to `automatically transpile and bundle dependencies`

Comment (DanielSchiavini) [reply apps/main/next.config.js]: shouldn't curve-lib be there, then? it's also in typescript, it wouldn't work without transpilation

File: apps/lend/tsconfig.json [hunk 3]
```diff
@@ -8,7 +8,7 @@
       "@/ui/*": ["../../../packages/ui/src/*"],
       "@/images": ["../../../packages/ui/src/images/index.ts"],
       "@/images/*": ["../../../packages/ui/src/images/*"],
-      "@/onboard": ["../../../packages/onboard-helpers/src/index.ts"],
+      "@/common/shared/features/*": ["../../../packages/curve-common/src/features/*"],
```
Comment (ignromanov) [inline apps/lend/tsconfig.json:11]:
```suggestion
      "@/common/*": ["../../../packages/curve-common/src/*"],
```
I think that's enough

File: apps/main/next.config.js (hunk 1 above)
Comment (ignromanov) [reply apps/main/next.config.js]: I guess yes, all the packages it can transpile

**Review by OnlyJousting — APPROVED** [review]

**Review by Macket — APPROVED** [review]

**Pull request merged**